File: MAP(Advanced Programming Methods)/countdownlatch/src/model/stmt/PrintStatement.java
package model.stmt;

import model.PrgState;
import model.expr.Expression;

public class PrintStatement implements Statement {

    private Expression expr;

    public PrintStatement(Expression e)
    {
        this.expr = e;
    }

    @Override
    public PrgState execute(PrgState p) {
        int res = this.expr.evaluate(p.getSymbolTable(),p.getHeap());
        p.getOutput().add(res);
        return null;
    }

    public String toString(){
        return "Print(" +expr.toString()+")";
    }
}

File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/Statements/PrintStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Model.Expressions;
using LabAssig7.Model;
namespace LabAssig7.Model.Statements
{
    public class PrintStatement:Statement
    {
        private Expression expr;

        public PrintStatement(Expression e)
        {
            this.expr = e;
        }

        public PrgState Execute(PrgState p)
        {
            int res = this.expr.Eval(p.Dict);
            p.Messages.AddValue(res);
            return p;
        }

        public override string ToString()
        {
            return "print(" + this.expr + ")\n";
        }
    }
}
